sbin/newfs_msdos/mkfs_msdos.c

```diff
@@ -566,9 +566,17 @@ mkfs_msdos(const char *fname, const char *dtype, const struct msdos_options *op)
     }
     print_bpb(&bpb);
     if (!o.no_create) {
-	gettimeofday(&tv, NULL);
-	now = tv.tv_sec;
-	tm = localtime(&now);
+	if (o.timestamp_set) {
+	    tv.tv_sec = now = o.timestamp;
```

Comment:
Indentation here differs from NetBSD, but this looks correct based on the style in the rest of the file. So, OK.